refarch-integrations/refarch-s3-integration/refarch-s3-integration-core/src/main/java/de/muenchen/refarch/s3/integration/adapter/in/rest/dto/FileDataDto.java

```diff
@@ -0,0 +1,27 @@
+package de.muenchen.refarch.s3.integration.adapter.in.rest.dto;
+
+import de.muenchen.refarch.s3.integration.adapter.in.rest.validation.FolderInFilePath;
+import de.muenchen.refarch.s3.integration.domain.model.FileData;
+import io.swagger.v3.oas.annotations.media.Schema;
+import jakarta.validation.constraints.Min;
+import jakarta.validation.constraints.NotEmpty;
+import jakarta.validation.constraints.NotNull;
+import jakarta.validation.constraints.Size;
+import lombok.Data;
+
+@Data
+@Schema(description = "File data for requesting a presigned url")
+public class FileDataDto {
```

Comment:
I suggest using Java Records for DTO objects instead of `@Data` annotated classes. This ensures immutability of objects via a native Java approach. Check other DTOs as well please. Won't comment any further.